src/test/test_cases.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from compress import CompressString
+
+
+@pytest.mark.parametrize("string, expected_result",
+                         [("aaabccccdd", "a3bc4d2"), ("aaaaaffffffffffc", "a5f10c"), ("abcd", "abcd"),
+                          ("ccceee12eccceee", "c3e4c3e3"), ("effeac01cb65c", "ef2eac2bc"), ("", ""),
+                          ("12345667890", "")])
+def test_use_cases(string, expected_result):
+	result = CompressString(string=string).compress()
+	assert result == expected_result
+
+
+@pytest.mark.parametrize("letter, expected_value", [('a', True), ('', False), ('1', False)])
+def test_is_letter(letter, expected_value):
+	result = CompressString(letter=letter)._is_letter()
+	assert result == expected_value
+
+
+@pytest.mark.parametrize("compression, letter, expected_value",
+                         [('a', 'a', True), ('', 'b', False), ('a1', 'a', False)])
+def test_is_previous_letter_same(compression, letter, expected_value):
+	result = CompressString(compression=compression, letter=letter)._is_previous_letter_same()
+	assert result == expected_value
```

Comment:
This a totally valid test so I'll just note that anything you can do to be clear about what's being tested is time well spent. The simplest thing you can do is add an inline doc like:
```python
def test_is_previous_letter_same(compression, letter, expected_value):
    """Check the sameness of the previous letter using known input values"""